tests/fetch_cc-test-reporter

```diff
@@ -0,0 +1,166 @@
+		trap 'cleanup 2>/dev/null || rm -f ${LOCK_FILE} 2>/dev/null > /dev/null || true ; wait ; exit 131 ;' SIGQUIT || EXIT_CODE=131
+		# SC2173 - https://github.com/koalaman/shellcheck/wiki/SC2173
+		#trap 'cleanup 2>/dev/null || rm -f ${LOCK_FILE} 2>/dev/null > /dev/null || true ; wait ; exit 1 ;' SIGSTOP || EXIT_CODE=7
+		trap 'cleanup 2>/dev/null || rm -f ${LOCK_FILE} 2>/dev/null > /dev/null || true ; wait ; exit 130 ;' SIGINT || EXIT_CODE=130
+		trap 'cleanup 2>/dev/null || rm -f ${LOCK_FILE} 2>/dev/null > /dev/null || true || true ; wait ; exit 137 ;' SIGABRT || EXIT_CODE=137
+		trap 'cleanup 2>/dev/null || rm -f ${LOCK_FILE} 2>/dev/null > /dev/null || true ; wait ; exit ${EXIT_CODE} ;' EXIT || EXIT_CODE=1
+		trap 'cleanup ; wait ; exit ${EXIT_CODE} ;' EXIT || EXIT_CODE=1
+else
+		echo "CodeCov already in progress by "`head ${LOCK_FILE}` ;
+		false ;
+		exit 126 ;
+fi
+
+# this is how test files are found:
+
+# THIS IS THE ACTUAL TEST DIR USED (update _TEST_ROOT_DIR as needed)
+_TEST_ROOT_DIR="./" ;
+if [[ -d ./.git ]] ; then
+	_TEST_ROOT_DIR="./" ;
+elif [[ -d ./tests ]] ; then
+	_TEST_ROOT_DIR="./" ;
+else
+	echo "FAIL: missing valid folder or file"
+	EXIT_CODE=1
+fi
+
+# This File  MUST BE GIT-IGNORED
+# to be SAFELY USED to store Tokens and env vars (update logic as needed)
+if [[ ( -r ./codecov_env ) ]] ; then
+	source ./codecov_env 2>/dev/null || true ;
+fi
+
+
+#########################
+# actual Work starts here
+#########################
+curl -fLso ./test-reporter-latest-${CI_OS:-linux}-amd64 https://codeclimate.com/downloads/test-reporter/test-reporter-latest-${CI_OS:-linux}-amd64 ;
+for i in 1 256 512 ; do
+	curl -fLso test-reporter-latest-${CI_OS:-linux}-amd64.SHA${i} "https://codeclimate.com/downloads/test-reporter/test-reporter-latest-${CI_OS:-linux}-amd64.SHA${i}" ; wait ;
+	curl -fLso test-reporter-latest-${CI_OS:-linux}-amd64.SHA${i}.sig "https://codeclimate.com/downloads/test-reporter/test-reporter-latest-${CI_OS:-linux}-amd64.SHA${i}.sig" ; wait ;
+	# test sha1/sha512 signatures if found and sha256 even if not found
+	if [[ ( -r test-reporter-latest-${CI_OS:-linux}-amd64.SHA${i} ) ]] || [[ ( ${i} -eq 256 ) ]] ; then
+		if [[ ( -r test-reporter-latest-${CI_OS:-linux}-amd64.SHA${i}.sig ) ]] ; then
+			# configure your CI evironment to trust the key at ????
+			# FP: KEY FP
+			# OR...
+			# Set CI=true to continue on missing keys
+			gpgv test-reporter-latest-${CI_OS:-linux}-amd64.SHA${i}.sig test-reporter-latest-${CI_OS:-linux}-amd64.SHA${i} || ${CI} || EXIT_CODE=126
+			rm -vf codecov.SHA${i}SUM.sig 2>/dev/null ;
+		fi
+		shasum -a $i -c --ignore-missing test-reporter-latest-${CI_OS:-linux}-amd64.SHA${i} || EXIT_CODE=126
+		rm -vf test-reporter-latest-${CI_OS:-linux}-amd64.SHA${i} 2>/dev/null ;
+	fi
+done
+
+if [[ ( ${EXIT_CODE} -eq 0 ) ]] ; then
+	mv -f test-reporter-latest-${CI_OS:-linux}-amd64 ./cc-test-reporter 2>/dev/null || EXIT_CODE=126
+	chmod -v 751 ./cc-test-reporter || EXIT_CODE=126
+fi
```

Comment:
- You misunderstand the nuance of the approach here. The sha-1, sha-256 and sha-512 hash-sums are ONLY relied on for data-integrity, not cryptographic significance, the cryptographic review (not even reliance, as it can even be overridden by setting "CI" to true) is instead on the GPG signature of the hash-sum, and checked BEFORE reliance regarding integrity (see line 142)

- also the failure of a sha-1 is still better than nothing if only the sha-1 hashsum is provided by the vendor. I feel leaving the sha-1 check is actually more pragmatic than removing it. Perhaps a better approach would be to warn and depreciate sha-1. Do you find a flaw with my reasoning?

Reply:
sure, let's explore your draft. Can you try to include a relevant CWE ID? I look forward to your draft either way.

Reply:
I'll include this with my next batch of changes shortly.